Definitions/Key

```diff
@@ -1,26 +1,30 @@
 class Key:
-    def __init__(self, symbol, startNote, isMajor):
-        this.symbol = symbol
-        this.startNote = startNote%12
-        this.isMajor = isMajor
+    def __init__(self, symbol, tonic, isMajor):
+        self.symbol = symbol
+        self.tonic = tonic%12
+        self.isMajor = isMajor
+        self.scale = this.getScale()
     
     def getScale(self):
-        scale = list()
-        counter = 0
-        if this.isMajor:
+        scale = dict{tonic:"1"}
+        counter = self.tonic
+        degreeCounter = 2
+        if self.isMajor:
             pattern = [2,2,1,2,2,2,1]
```

Comment:
I don't think you'll need the last item in the array as it will just repeat the tonic at a new octave.